app/assets/javascripts/uploads.js

```diff
@@ -26,6 +27,27 @@
     }
   }
 
+  Danbooru.Upload.initialize_submit = function() {
+    $("#submit-button").click(function(e) {
+      var error_messages = [];
+      if (($("#upload_file").val() === "") && ($("#upload_source").val() === "")) {
+        error_messages.push("Must choose file or specify source");
+      }
+      if (!$("#upload_rating_s")[0].checked && !$("#upload_rating_q")[0].checked && !$("#upload_rating_e")[0].checked) {
```

Comment:
It's possible to specify the rating in the tag string using the `rating:` metatag. This case would have to be checked too.